src/lib/plugins/addPagination.ts

```diff
@@ -27,18 +34,27 @@ export const createPageStore = ({ items, initialPageSize, initialPageIndex }: Pa
 	};
 	const setPageSize = (newPageSize: number) => updatePageSize(() => newPageSize);
 
-	const pageCount = derived([pageSize, items], ([$pageSize, $items]) => {
-		const $pageCount = Math.ceil($items.length / $pageSize);
+	const pageIndex = writable(initialPageIndex);
+
+	let pageCount;
+	function calcPageCount([$pageSize, $itemsCount]: [$pageSize: number, $itemsCount: number]) {
```

Comment:
I didn't name the derive function before, so the introduction of a name helps!

A potential improvement would be to make it clear that this doesn't only calculate the page count, but also introduces side effects for the `pageIndex` store.

Maybe we can name this `calcPageCountAndLimitIndex`? I'm not sure about the name exactly, but it would help prevent any misunderstandings about the function's purpose.

Perhaps we